politeiad/cache/testcache/decred.go

```diff
@@ -0,0 +1,127 @@
+// Copyright (c) 2017-2019 The Decred developers
+// Use of this source code is governed by an ISC
+// license that can be found in the LICENSE file.
+
+package testcache
+
+import (
+	decred "github.com/decred/politeia/decredplugin"
+	"github.com/decred/politeia/politeiad/cache"
+)
+
+func (c *testcache) getComments(payload string) (string, error) {
+	gc, err := decred.DecodeGetComments([]byte(payload))
+	if err != nil {
+		return "", err
+	}
+
+	c.RLock()
```

Comment:
why is this locked?

Reply:
Because of the cache read a couple of lines below it.

https://github.com/decred/politeia/pull/755/files/16f88eabd4c933098ba20a733917d2d5996090c0#diff-3bc46c268ad664a05b40e36e7f531831R23